app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/viewmodel/MoviesDetailsViewModel.kt

```diff
@@ -4,27 +4,37 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.bashkevich.androidfundamentals.data.JsonLoad
-import com.bashkevich.androidfundamentals.data.Movie
+import com.bashkevich.androidfundamentals.model.EntityMapper
+import com.bashkevich.androidfundamentals.model.MoviesRepository
+import com.bashkevich.androidfundamentals.model.RetrofitModule
+import com.bashkevich.androidfundamentals.model.entity.Actor
+import com.bashkevich.androidfundamentals.model.entity.Movie
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 
 class MoviesDetailsViewModel(
-        private val jsonLoad: JsonLoad
+    private val moviesRepository: MoviesRepository
 ) : ViewModel() {
     private var _movieLiveData = MutableLiveData<Movie>()
 
     val movieLiveData: LiveData<Movie>
         get() = _movieLiveData
 
-    fun getMovieFromList(movieId: Int) {
-        if (_movieLiveData.value?.id != movieId) {
+    fun loadMovie(movie: Movie) {
+        if (_movieLiveData.value?.id != movie.id) {
             viewModelScope.launch {
-                val movies = jsonLoad.loadMovies()
-                val movie = movies.find { movie -> movie.id == movieId }
+
+                movie.actors = addActorsToMovie(movie.id)
```

Comment:
Why not to get movie by id as it was done before?

Reply:
I load it when I get list of movies and not to do same thing twice I just get Movie once? when I load a list, and then I put Parcelable into details

Reply:
It is technically caching of movie. But caching is task not for the UI layer but for the lower layer - repository or view model/navigation